src/mb_caa_dimensions/InfoCache.ts

```diff
@@ -0,0 +1,268 @@
+        try {
+            await this.db.put(storeName, {
+                ...value,
+                addedDatetime: Date.now(),
+            }, imageUrl);
+            LOGGER.debug(`Successfully stored ${imageUrl} in cache`);
+        } catch (e) /* istanbul ignore next: Difficult to cover */ {
+            LOGGER.error(`Failed to store ${imageUrl} in cache`, e);
+        }
+    }
+
+    async getDimensions(imageUrl: string): Promise<Dimensions | undefined> {
+        return this.get(CACHE_DIMENSIONS_STORE_NAME, imageUrl);
+    }
+
+    async putDimensions(imageUrl: string, dimensions: Dimensions): Promise<void> {
+        return this.put(CACHE_DIMENSIONS_STORE_NAME, imageUrl, dimensions);
+    }
+
+    async getFileInfo(imageUrl: string): Promise<FileInfo | undefined> {
+        return this.get(CACHE_FILE_INFO_STORE_NAME, imageUrl);
+    }
+
+    async putFileInfo(imageUrl: string, fileInfo: FileInfo): Promise<void> {
+        return this.put(CACHE_FILE_INFO_STORE_NAME, imageUrl, fileInfo);
+    }
+}
+
+async function maybePruneDb(db: IDBPDatabase<CacheDBSchema>): Promise<void> {
+    const lastPruneTimestamp = parseInt(localStorage.getItem(CACHE_TIMESTAMP_NAME) ?? '0');
+    const timeSinceLastPrune = Date.now() - lastPruneTimestamp;
+    if (timeSinceLastPrune < CACHE_CHECK_INTERVAL) {
+        LOGGER.debug(`Cache was last pruned at ${new Date(lastPruneTimestamp)}, pruning is unnecessary`);
+        return;
+    }
+
+    LOGGER.info('Pruning stale entries from cache');
+    const pruneRange = IDBKeyRange.upperBound(Date.now() - CACHE_STALE_TIME);
+
+    async function iterateAndPrune(storeName: StoreNames<CacheDBSchema>): Promise<void> {
+        let cursor = await db.transaction(storeName, 'readwrite').store.index(CACHE_ADDED_TIMESTAMP_INDEX).openCursor(pruneRange);
+        while (cursor !== null) {
+            LOGGER.debug(`Removing ${cursor.key} (added at ${new Date(cursor.value.addedDatetime)})`);
+            await cursor.delete();
+            cursor = await cursor.continue();
+        }
+    }
+
+    await iterateAndPrune(CACHE_DIMENSIONS_STORE_NAME);
+    await iterateAndPrune(CACHE_FILE_INFO_STORE_NAME);
+
+    LOGGER.debug('Done pruning stale entries');
+    localStorage.setItem(CACHE_TIMESTAMP_NAME, Date.now().toString());
+}
+
+
+//// Legacy stuff
+
+interface CacheDBSchemaV1 extends DBSchema {
+    'cacheStore': {
+        key: string;
+        value: {
+            url: string;
+            width: number;
+            height: number;
+            size: number;
+            format: string;
+            added_datetime: number;
+        };
+    };
+}
+
+type Migrator = (db: IDBPDatabase<CacheDBSchema>, tx: IDBPTransaction<CacheDBSchema, Array<StoreNames<CacheDBSchema>>, 'versionchange'>) => Promise<void>;
+/**
+ * Map of database migrations, indexed by current version.
+ */
+const DbMigrations: Record<number, Migrator> = {
```

Comment:
This looks like an overkill to me, but since you've already written the code, there is no reason not to keep it 🤯

Reply:
It was simpler in the beginning 😅 I don't plan on doing many DB migrations anymore though.